File: lab4 mpp/ProjectNetworking/DonatieDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using teledon.domain;

namespace teledon.networking.dto
{
    [Serializable]
    public class DonatieDTO
    {
        public DonatorDTO donator;
        public CazDTO caz;
        public int suma;

        public DonatieDTO(CazCaritabil c, Donator d, int s)
        {
            this.caz = DTOUtils.getDTO(c);
            this.donator = DTOUtils.getDTO(d);
            this.suma = s;
        }
        public DonatieDTO(CazDTO c, DonatorDTO d, int s)
        {
            this.caz = c;
            this.donator = d;
            this.suma = s;
        }

        public virtual DonatorDTO Donator
        {
            get
            {
                return donator;
            }
            set
            {
                this.donator = value;
            }
        }

        public virtual CazDTO Caz
        {
            get
            {
                return caz;
            }
            set
            {
                this.caz = value;
            }
        }

        public virtual int Suma
        {
            get
            {
                return suma;
            }
            set
            {
                this.suma = value;
            }
        }

    }
}

File: GUImpp/TeledonNetworking/src/main/java/teledon/network/dto/DonatieDTO.java
package teledon.network.dto;

import teledon.model.CazCaritabil;
import teledon.model.Donator;

import java.io.Serializable;

public class DonatieDTO implements Serializable {
    public Donator donator;
    public CazCaritabil caz;
    public int suma;

    @Override
    public String toString() {
        return "teledon.model.Donatie{ donator=" + donator +
                ", caz=" + caz +
                ", suma=" + suma +
                '}';
    }

    public DonatieDTO(Donator donator, CazCaritabil caz, int suma) {
        this.donator = donator;
        this.suma = suma;
        this.caz = caz;
    }
    public DonatieDTO() {}
    public CazCaritabil getCaz() {
        return this.caz;
    }

    public Donator getDonator() {
        return donator;
    }

    public void setDonator(Donator donator) {
        this.donator = donator;
    }

    public int getSuma() {
        return suma;
    }

    public void setSuma(int suma) {
        this.suma = suma;
    }
}
